slp-postage-protocol.md

```diff
@@ -0,0 +1,268 @@
+| 3 | 546 satoshis *stamp* | **(20 SLP tokens)** 546 satoshis  *change* |
+| 4 | 546 satoshis *stamp* | |
+
+AFTER POSTAGE (Multiple SLP Inputs / Varying Stamp Weights):</br>
+
+| INDEX | INPUT | OUTPUT |
+| ------------ | ------------ | ------------------------------------------|
+| 0 | **(10 SLP tokens)** 546 satoshis | **SLP OP_RETURN** |
+| 1 | **(35 SLP tokens)** 546 satoshis | **(20 SLP tokens)** 546 satoshis |
+| 2 | **(25 SLP tokens)** 546 satoshis | **(30 SLP tokens)** 546 satoshis  *paying postage* |
+| 3 | 911 satoshis *stamp* | **(20 SLP tokens)** 546 satoshis  *change* |
+
+
+### MIME Types
+
+The MIME (RFC 2046) Media Types follow the basic specification described in [Simple Ledger Payment Protocol](https://github.com/simpleledger/slp-specifications/blob/master/slp-payment-protocol.md)
+
+| Message  | Type/Subtype                                  |
+| ------------ | ------------------------------------------|
+| Payment      | application/simpleledger-payment          |
+| PaymentACK      | application/simpleledger-paymentack    |
+
+#### Example
+
+A wallet client generating a Payment message would precede the binary message data with the following headers:
+
+``Content-Type: application/simpleledger-payment``</br>
+``Accept: application/simpleledger-paymentack``</br>
+``Content-Transfer-Encoding: binary``</br>
+
+
+### Common Errors
+
+| HTTP Status Code | Response | Cause |
+|---|---|---|
+| 400 | Unsupported Content-Type for payment | Your Content-Type header was not valid |
+| 400 | Invalid Payment | The payment is would be invalid even if postage was added |
+| 400 | Could not parse OP_RETURN output | SLP OP_RETURN output is malformed |
+| 400 | Unsupported SLP token | The SLP token in the transaction is not among the supported tokens |
+| 400 | Insufficient postage paid | The amount paid to the post office address is insufficient to cover postage |
+| 400 | Stamps currently unavailable. In need of refill | The post office has insufficient stamp UTXOs available. Please inform post office support |
+| 500 | Error | An error has occurred on the post office server |
+
+
+### Notes On Privacy And Censorship
+
+In a case where a post office server cannot determine the nature or source of a transaction, the sender's privacy is maintained. There is also little incentive for a post office to refuse any given transaction. However, in the case where a post office determines that a transaction benefits a (business) competitor, they may perceive greater benefit in not adding postage and refusing the transaction than would be received from the revenue gained by providing postage. To follow are some optional, but beneficial practices wallets can follow to keep incentives aligned between sender and post office.
+
+1. Non-reuse of addresses by senders and third parties such as payment protocol servers. This prevents potential refusal of transactions based on benefit to business competitors.
```

Comment:
This sort of gets to my point above about, when and why the "address" will be changed?  If wallets are just querying for general post-office support so that the wallet can "shop" for the best postage rate, then will that cause the addresses given out by the server to change.  That would then result in many addresses being handed out to clients which would not get used.